cpp/src/strings/contains.cu

```diff
@@ -0,0 +1,244 @@
+ * Small to medium instruction lengths can use the stack effectively though smaller executes faster.
+ * Longer patterns require global memory.
+ *
+ */
+template<size_t stack_size>
+struct contains_fn
+{
+    reprog_device prog;
+    column_device_view d_strings;
+    bool bmatch{false}; // do not make this a template parameter to keep compile times down
+
+    __device__ cudf::experimental::bool8 operator()(size_type idx)
+    {
+        if( d_strings.is_null(idx) )
+            return 0;
+        u_char data1[stack_size], data2[stack_size];
+        prog.set_stack_mem(data1,data2);
+        string_view d_str = d_strings.element<string_view>(idx);
+        int32_t begin = 0;
+        int32_t end = bmatch ? 1 : d_str.length(); // 1=match only the beginning of the string
+        return static_cast<experimental::bool8>(prog.find(idx,d_str,begin,end));
+    }
+};
+
+//
+std::unique_ptr<column> contains_util( strings_column_view const& strings,
+                                       std::string const& pattern,
+                                       bool beginning_only = false,
+                                       rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+                                       cudaStream_t stream = 0)
+{
+    auto strings_count = strings.size();
+    auto strings_column = column_device_view::create(strings.parent(),stream);
+    auto d_column = *strings_column;
+
+    // compile regex into device object
+    auto prog = reprog_device::create(pattern,get_character_flags_table(),strings_count,stream);
+    auto d_prog = *prog;
+
+    // create the output column
+    auto results = make_numeric_column( data_type{BOOL8}, strings_count,
+        copy_bitmask( strings.parent(), stream, mr), strings.null_count(), stream, mr);
+    auto d_results = results->mutable_view().data<cudf::experimental::bool8>();
+
+    // fill the output column
+    auto execpol = rmm::exec_policy(stream);
+    int regex_insts = d_prog.insts_counts();
+    if( (regex_insts > MAX_STACK_INSTS) || (regex_insts <= RX_SMALL_INSTS) )
+        thrust::transform(execpol->on(stream),
+            thrust::make_counting_iterator<size_type>(0),
+            thrust::make_counting_iterator<size_type>(strings_count),
+            d_results, contains_fn<RX_STACK_SMALL>{d_prog, d_column, beginning_only} );
+    else if( regex_insts <= RX_MEDIUM_INSTS )
+        thrust::transform(execpol->on(stream),
+            thrust::make_counting_iterator<size_type>(0),
+            thrust::make_counting_iterator<size_type>(strings_count),
+            d_results, contains_fn<RX_STACK_MEDIUM>{d_prog, d_column, beginning_only} );
+    else
+        thrust::transform(execpol->on(stream),
+            thrust::make_counting_iterator<size_type>(0),
+            thrust::make_counting_iterator<size_type>(strings_count),
+            d_results, contains_fn<RX_STACK_LARGE>{d_prog, d_column, beginning_only} );
+
+    results->set_null_count(strings.null_count());
```

Comment:
Why should be set null_count twice ?

Reply:
Calling `mutable_view()` on a `column` invalidates the `null_count`. This sets it back correctly.

Reply:
In that case we can just pass UNKNOWN in earlier `make_numeric_column` and set the null_count over here.

Reply:
The null-count is necessary here for the `copy_bitmask` call to work correctly.